### PR TITLE
`<ModalPreviewLayout/>`- adding tooltip props for close button & navigation buttons

### PR DESCRIPTION
**Pull request by TalZa1810**
### 🔦 Summary
1. Having the option to custom the tooltips of the close button & navigation buttons.
2. Fixing issue with tooltip flakiness when clicking on prev & next buttons.

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [x] Component
  - [ ] Visual test

### CODE REVIEW

File: src/ModalPreviewLayout/NavigationButton/NavigationButton.js [hunk 1]
```diff
@@ -19,27 +19,35 @@ const tooltipProps = {
...
-    <Tooltip {...tooltipProps[direction]}>
-      <IconButton as="button" skin="transparent" {...props}>
-        {iconButtonArrow[direction]}
-      </IconButton>
-    </Tooltip>
-  </div>
-);
+const NavigationButton = ({ direction, tooltipText, ...props }) => {
+  //TODO: adding tests for checking tooltip text
+  if (typeof tooltipText !== 'string') {
```
Comment (shlomitc) [inline src/ModalPreviewLayout/NavigationButton/NavigationButton.js]: Same here

**Comment by TalZa1810** [comment]
Hi @EgoziE ,
Do you think that I should also visual test each of the tooltip props? WDYT?

File: src/ModalPreviewLayout/ModalPreviewLayout.js [hunk 2]
```diff
@@ -25,10 +25,19 @@ class ModalPreviewLayout extends React.PureComponent {
...
+    prevButtonTooltipText: PropTypes.string,
+    /** Tooltip next button text */
+    nextButtonTooltipText: PropTypes.string,
   };
 
   static defaultProps = {
     shouldCloseOnOverlayClick: true,
+    closeButtonTooltipText: 'Close',
+    prevButtonTooltipText: 'Previous',
+    nextButtonTooltipText: 'Next',
```
Comment (EgoziE) [inline src/ModalPreviewLayout/ModalPreviewLayout.js]: Shlomi raised a point an another PR, adding default values for "user props" is not recommended.
For example, if I don't want a tooltip, I'm getting the default, which is not even translated.

I think these 3 lines should be deleted 🗡

**Comment by TalZa1810** [comment]
Hi @EgoziE ,
Can you please review the changes? I've removed the default props and updated the code and unit tests accordingly. Also, after merging and releasing, the types PR can be merged:
https://github.com/wix/wix-ui/pull/1756

FYI @michalavraham

File: src/ModalPreviewLayout/ModalPreviewLayout.js [hunk 3]
```diff
@@ -55,26 +61,40 @@ class ModalPreviewLayout extends React.PureComponent {
     };
   }
 
-  _renderNavigationButtons() {
+  _renderNavigationButtons(hasLeft, hasRight) {
+    const { prevButtonTooltipText, nextButtonTooltipText } = this.props;
```
Comment (mosheKerbel) [inline src/ModalPreviewLayout/ModalPreviewLayout.js]: looks like it's not up to the user if nav buttons should be rendered
in this case, shouldn't `prevButtonTooltipText` & `nextButtonTooltipText` be mandatory props?

Comment (mosheKerbel) [reply src/ModalPreviewLayout/ModalPreviewLayout.js]: my bad, i see it's part of `NavigationButton`

Comment (TalZa1810) [reply src/ModalPreviewLayout/ModalPreviewLayout.js]: If they are mandatory, we should have default props for this, no?
That's why I changed the logic: 
https://github.com/wix/wix-style-react/pull/4965#discussion_r376956592

should we make them mandatory and have default props for this or not?

Comment (TalZa1810) [reply src/ModalPreviewLayout/ModalPreviewLayout.js]: or just have them required without default props?

**Pull request merged**